app/code/Magento/Sales/Model/OrderRepository.php

```diff
@@ -140,6 +157,12 @@ public function get($id)
                     __("The entity that was requested doesn't exist. Verify the entity and try again.")
                 );
             }
+            $date = $this->locale->date($entity->getCreatedAt());
```

Comment:
Hi, @webkul-ratnesh. Thank you for your contribution!
Well, you are right - this solution will solve the problem. However, I don't really think a repository is a place where the date formatting should happen, - it's a part of presentation layer logic, but not the service one.

In this case, I would suggest moving this logic into something kind of after-plugin that applies for `webapi_rest` and `webapi_soap` areas only.